lib/puppet/indirector/resource/ral.rb

```diff
@@ -12,7 +12,7 @@ def allow_remote_requests?
   def find( request )
     # find by name
     res   = type(request).instances.find { |o| o.name == resource_name(request) }
-    res ||= type(request).new(:name => resource_name(request), :audit => type(request).properties.collect { |s| s.name })
+    res ||= type(request).new(:name => resource_name(request))
```

Comment:
@joshcooper helped me walk through this one, including manual testing with the `puppet resource` command using `file`, which goes through this code path. We were not able to make it break. Do you have a concrete example of a case where this could cause trouble? (my grasp on how this stuff all works is still sorta weak)